modules/libcom/src/osi/os/RTEMS/osdTime.cpp

```diff
@@ -21,28 +23,37 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <string.h>
+#if __RTEMS_MAJOR__ < 5
 #include <rtems/rtems_bsdnet_internal.h>
+#endif
 #include "epicsTime.h"
 #include "osdTime.h"
 #include "osiNTPTime.h"
 #include "osiClockTime.h"
 #include "generalTimeSup.h"
 
 extern "C" {
-
 extern rtems_interval rtemsTicksPerSecond;
+#ifdef RTEMS_LEGACY_STACK
 int rtems_bsdnet_get_ntp(int, int(*)(), struct timespec *);
 static int ntpSocket = -1;
+#endif
+
 
 void osdTimeRegister(void)
 {
     /* Init NTP first so it can be used to sync ClockTime */
+#ifdef RTEMS_LEGACY_STACK
     NTPTime_Init(100);
     ClockTime_Init(CLOCKTIME_SYNC);
-
+#else
+printf(" OsdTimeRegister Not implemented yet for new libbsd ...\n");
+#endif
```

Comment:
void osdTimeRegister(void)
{
    /* Init NTP first so it can be used to sync ClockTime */
#ifdef RTEMS_LEGACY_STACK
    NTPTime_Init(100);
    ClockTime_Init(CLOCKTIME_SYNC);
#else
printf(" osdTimeRegister: not implemented for non RTEMS_LEGACY_STACK \n");
#endif
    osdMonotonicInit();
}